calcom/cal.com
# chore: [app-router-migration 27] migrate the "signup" page

## What does this PR do?
- This PR migrates the `/signup` page to the app directory (which runs under the App Router).
- The migrated page exists under `/future/signup` and is accessible at all times.
- We support a redirect from the former to the latter with the middleware and environmental variables, as done previously.

## Type of change

## How should this be tested?
There should be no changes between the `/signup` and `/future/signup` in terms of performance and content (on Lighthouse both pages get near 100 scores).

If the ```APP_ROUTER_SIGNUP_ENABLED``` is set to `1` and ```AB_TEST_BUCKET_PROBABILITY``` is set to `100`, then the future signup page should be rendered in place of the `/signup` page WITHOUT a visible redirect. We can see it by inspecting the existence of the `x-calcom-future-routes-enabled` cookie.

## Mandatory Tasks
